### PR TITLE
Adds support for catalog name in API

### PR DESCRIPTION
**Pull request by SM43**
This adds catalog name in url for APIs:
```
-/resource/<catalog>/<kind>
-/resource/<catalog>/<kind>/<version>
```
# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: api/design/resource.go [hunk 1]
```diff
@@ -140,21 +140,22 @@ var _ = Service("resource", func() {
 		})
 	})
 
-	Method("ByKindName", func() {
-		Description("Find resources using name and kind")
+	Method("ByCatalogKindName", func() {
+		Description("Find resources using name of catalog & name and kind of resource")
```
Comment (PuneetPunamiya) [inline api/design/resource.go:144]: Can we change the description to `Find resources using name of catalog, resource name and kind of resource`

File: api/pkg/service/resource/resource_test.go [hunk 2]
```diff
@@ -210,8 +210,19 @@ func TestByKindNameVersion_NoResourceWithName(t *testing.T) {
...
 	resourceSvc := New(tc)
-	payload := &resource.ByKindNameVersionPayload{Kind: "task", Name: "foo", Version: "0.1"}
-	_, err := resourceSvc.ByKindNameVersion(context.Background(), payload)
+	payload := &resource.ByCatalogKindNameVersionPayload{Catalog: "catalog-official", Kind: "task", Name: "foo", Version: "0.1"}
+	_, err := resourceSvc.ByCatalogKindNameVersion(context.Background(), payload)
+	assert.Error(t, err)
+	assert.EqualError(t, err, "Resource not found")
+}
+
+func TestByKindNameVersion_NoCatalogWithName(t *testing.T) {
```
Comment (PuneetPunamiya) [inline api/pkg/service/resource/resource_test.go:219]: nit: Can we just change this name to `TestByKindNameVersion_WithNoCatalogName`

**Comment by PuneetPunamiya** [comment]
@SM43  Thank you for the PR, the patch looks overall good to me only some nit changes to address

**Review by sthaha — COMMENTED** [review]
/lgtm
/approve
/meow

**Pull request merged**